### DISCUSSION TITLE
Change channels(mppt) name on mqtt

### DISCUSSION THREAD

**Discussion by Ayo91** [tbnobody/OpenDTU#1657] [Ideas]
I have changed the different names on the UI (string name) because I wanted a name that matched with my setup.

However, I still see the old names on Home Assistant (ch1, ch2, etc). Is there any possibility to setup custom names also for the mqtt integration?
![Uploading Screenshot_2024-01-17-12-44-14-726_io.homeassistant.companion.android.jpg…]()

**Comment by NintendoTito** [comment]
I can understand the problem, I have a similar problem. However, I believe that this can be solved by editing the config.yaml within Home Assistant.

You proceed in a similar way when you maintain the measurement of electricity consumption and feed-in in Home Assistant. I think you can adapt the name received via MQTT within HA accordingly for a more meaningful display.

**Reply by Ayo91** [reply]
Thanks for your feedback.

I asked it because I have not any manual config in my config.yaml and lucky me... its all automatically setup thanks to the discovery.

I also know how to set friendly names but I would like avoid to do that in almost 100 entities... And I will have more in the future...

Each string in OpenDtu generate several info like voltage, current, power... and all of them with the default name. 4 strings x 3 entities... you can imagine because I have 6 inverters right now....

It will be perfect that if you set the names of the strings in the openDtu UI is also propagated.

Do you understand it?

**Reply by Ayo91** [reply]
![Screenshot_2024-01-17-20-25-47-895_com android chrome](https://github.com/tbnobody/OpenDTU/assets/2479896/de0a40e5-60c0-4a9e-aeab-48b5e577f8f7)

**Comment by NintendoTito** [comment]
@Ayo91 I think this is the solution, just tested it:

https://www.home-assistant.io/docs/configuration/customizing-devices/

You have to call up the entity and go to the settings via the cogwheel. In my case, no name was assigned there, so the value sent via MQTT is used.

I then entered a "coherent" name. And the name is drawn directly both in the Energy Dashboard and as the entity name in the listing.
